dot/types/equivocation.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package types
+
+// BabeEquivocationProof represents a babe equivocation proof.
+// An equivocation happens when a validator produces more than one block on the same slot.
+// The proof of equivocation are the given distinct headers that were signed by the validator
+// and which include the slot number.
+type BabeEquivocationProof struct {
+	// Offender is the public key of the equivocator.
+	Offender AuthorityID
+	// Slot at which the equivocation happened.
+	Slot uint64
+	// FirstHeader is the first header involved in the equivocation.
+	FirstHeader Header
+	// SecondHeader is the second header involved in the equivocation.
+	SecondHeader Header
+}
+
+// AuthorityID represents a babe authority identifier.
+type AuthorityID [32]byte
+
+// OpaqueKeyOwnershipProof is an opaque type used to represent the key ownership proof at the
+// runtime API boundary. The inner value is an encoded representation of the actual key
+// ownership proof which will be parameterized when defining the runtime. At
+// the runtime API boundary this type is unknown and as such we keep this
+// opaque representation, implementors of the runtime API will have to make
```

Comment:
Is this part of the comment not rust specific? Doesnt golang do type checking on the underlying type? Example, you could pass this as a para to a function that takes []byte as an argument and this would compile right?

Other thing is, do we need this type? Since its just a byte wrapper and doesn't provide (I think) the type checking benefits that rust does, is it necessary? If its just for readability and to mimick substrate I am okay with that, but if its not serving a good purpose I am okay with replacing the type with relevant variable names

Reply:
This comment is copied from the substrate code.  I don't mind defining types that don't have receiver methods.  It's the same on the substrate side.  Maybe we need a constructor for this?

Reply:
@jimjbrettj This is mostly for readability and to add some extra info about what key ownership proof is. I could just replace this with []byte and things would work just fine.

> Maybe we need a constructor for this?

Seems unnecessary. I could wrap bytes with `OpaqueKeyOwnershipProof([]byte{})` or use `NewOpaqueKeyOwnershipProof([]byte{})`. Both seems like pretty much the same thing.